drivers/ethernet/eth_lan9250.c

```diff
@@ -0,0 +1,713 @@
+/* LAN9250 Stand-alone Ethernet Controller with SPI
+ *
+ * Copyright (c) 2024 Mario Paja
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT microchip_lan9250
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/ethernet.h>
+#include <ethernet/eth_stats.h>
+
+#include "eth_lan9250_priv.h"
+
+LOG_MODULE_REGISTER(eth_lan9250, CONFIG_ETHERNET_LOG_LEVEL);
+
+static int lan9250_write_sys_reg(const struct device *dev, uint16_t address, uint32_t data)
+{
+	const struct lan9250_config *config = dev->config;
+	uint8_t cmd[1] = {LAN9250_SPI_INSTR_WRITE};
+	uint8_t addr[2];
+	uint8_t instr[4];
+	struct spi_buf tx_buf[3];
+	const struct spi_buf_set tx = {.buffers = tx_buf, .count = 3};
+
+	sys_put_be16(address, addr);
+	sys_put_le32(data, instr);
+
+	tx_buf[0].buf = &cmd;
+	tx_buf[0].len = ARRAY_SIZE(cmd);
+	tx_buf[1].buf = addr;
+	tx_buf[1].len = ARRAY_SIZE(addr);
+	tx_buf[2].buf = instr;
+	tx_buf[2].len = ARRAY_SIZE(instr);
+
+	return spi_write_dt(&config->spi, &tx);
+}
+
+static int lan9250_read_sys_reg(const struct device *dev, uint16_t address, uint32_t *value)
+{
+	const struct lan9250_config *config = dev->config;
+	uint8_t cmd[1] = {LAN9250_SPI_INSTR_READ};
+	uint8_t addr[2];
+	struct spi_buf tx_buf[3];
+	struct spi_buf rx_buf[3];
+	const struct spi_buf_set tx = {.buffers = tx_buf, .count = 3};
+	const struct spi_buf_set rx = {.buffers = rx_buf, .count = 3};
+
+	sys_put_be16(address, addr);
+
+	tx_buf[0].buf = &cmd;
+	tx_buf[0].len = ARRAY_SIZE(cmd);
+	tx_buf[1].buf = addr;
+	tx_buf[1].len = ARRAY_SIZE(addr);
+	tx_buf[2].buf = NULL;
+	tx_buf[2].len = sizeof(uint32_t);
+
+	rx_buf[0].buf = NULL;
+	rx_buf[0].len = 1;
+	rx_buf[1].buf = NULL;
+	rx_buf[1].len = 2;
+	rx_buf[2].buf = value;
+	rx_buf[2].len = sizeof(uint32_t);
+
+	return spi_transceive_dt(&config->spi, &tx, &rx);
+}
+
+static int lan9250_wait_ready(const struct device *dev, uint16_t address, uint32_t mask,
+			      uint32_t expected, uint32_t m_second)
+{
+	uint32_t tmp;
+	int wait_time = 0;
+
+	while (true) {
```

Comment:
Any reason reason for not using the `lan9250_wait_ready()` function here?

Reply:
I am sorry but I don't understand the question because this is the `lan9250_wait_ready()` function

Reply:
oops. apologies. This should comment have been in the `lan9250_spi_check()`. I can try to rephrase to make it clear. 

The `lan9250_spi_check()` can use the `lan9250_wait_ready()` instead of its own do while loop in https://github.com/zephyrproject-rtos/zephyr/pull/77035/files#diff-90c0053462ceca8a085a606863af422c5a7f56ac4012c094db95c6b44267f0a2R230-R233

Reply:
In that case, I think you are right, `lan9250_spi_check()` could use `lan9250_wait_ready()` to wait for the SPI setup